### PR TITLE
refactor: Remove usages of reactable from TableLoader

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
The goal is to replace all uses of `reactable-arc` library with DataTable from superset-ui (which uses `react-table` under the hood) and keep the UI/UX mostly unchanged. The reason for that is that `reactable-arc` has been unsupported for 2 years now, uses deprecated lifecycle methods and is incompatible with Typescript.
This PR replaces `reactable-arc` in `TableLoader` component.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before (nothing renders here, even though the data is correctly fetched. Bug perhaps?):

image

After:

image

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by kgabryje** [comment]
@rusackas Can you please take a look?

File: superset-frontend/src/components/TableLoader.jsx [hunk 1]
```diff
@@ -71,40 +99,26 @@ class TableLoader extends React.PureComponent {
...
-          </Tr>
-        ))}
-      </Table>
+      <ListView
+        columns={columns.map(column => ({
+          accessor: column,
+          Header: column,
+        }))}
+        data={this.state.data}
+        count={this.state.data.length}
```
Comment (eschutho) [inline superset-frontend/src/components/TableLoader.jsx:117]: is this count supposed to come from the api as a total count?

File: superset-frontend/src/components/ListView/ListView.tsx [hunk 2]
```diff
@@ -48,8 +48,24 @@ const ListViewStyles = styled.div`
...
+        }
+        &:horizontal {
+          height: 7px;
+        }
+      }
+
+      &::-webkit-scrollbar-thumb {
+        border-radius: 4px;
+        background-color: rgba(0, 0, 0, 0.5);
+        -webkit-box-shadow: 0 0 1px rgba(255, 255, 255, 0.5);
```
Comment (rusackas) [inline superset-frontend/src/components/ListView/ListView.tsx:67]: Aha! It's in Emotion now! I made some comments on https://github.com/apache/incubator-superset/pull/10982 about these color/pixel values, and how they might be updated to leverage the Theme. If those values are used here, it seems that other PR might not need all the LESS code at all?

Comment (rusackas) [reply superset-frontend/src/components/ListView/ListView.tsx:67]: Happy to copy over the theme bits as "suggestions" here so you can directly commit them, if it helps.

File: superset-frontend/src/components/TableLoader.jsx [hunk 3]
```diff
@@ -44,9 +43,42 @@ class TableLoader extends React.PureComponent {
     };
   }
 
-  UNSAFE_componentWillMount() {
```
Comment (rusackas) [inline superset-frontend/src/components/TableLoader.jsx:47]: Yaaaaay! Happy to see these go away :)

**Review by rusackas — CHANGES_REQUESTED** [review]
LGTM in general (this is a great PR!), but would love to see the Emotion tweaks to use Theme variables wherever possible, and curious about @eschutho 's question

File: superset-frontend/src/components/ListView/ListView.tsx (hunk 2 above)
Comment (kgabryje) [reply superset-frontend/src/components/ListView/ListView.tsx]: Done

File: superset-frontend/src/components/TableLoader.jsx (hunk 1 above)
Comment (eschutho) [reply superset-frontend/src/components/TableLoader.jsx:117]: Ok, sounds good. 👍

**Review by rusackas — APPROVED** [review]
LGTM... just needs a rebase and I (or whoever!) can merge

**Comment by kgabryje** [comment]
@rusackas Thanks for review. However, due to [this](https://github.com/apache/incubator-superset/issues/11142) issue, which is related to changes made here, what do you think about postponing merging this PR until the issue is resolved? So that we don't risk another regression.

**Pull request closed without merging**